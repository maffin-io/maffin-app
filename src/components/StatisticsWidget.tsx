import React from 'react';

export type StatisticsWidgetProps = {
  title: string,
  stats: string,
  description: string,
  className?: string,
  statsTextClass?: string,
};

export default function StatisticsWidget({
  title,
  stats,
  description,
  className = '',
  statsTextClass = '',
}: StatisticsWidgetProps): JSX.Element {
  return (
    <div className={`${className} bg-gunmetal-700 rounded-sm p-6`}>
      <p>
        {title}
      </p>
      <p className={`${statsTextClass} text-2xl font-semibold my-3`}>{stats}</p>

      <span className="text-sm">
        {description}
      </span>
    </div>
  );
}
