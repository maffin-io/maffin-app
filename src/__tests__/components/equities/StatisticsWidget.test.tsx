import React from 'react';
import { render } from '@testing-library/react';

import StatisticsWidget from '@/components/StatisticsWidget';

describe('StatisticsWidget', () => {
  it('renders as expected', () => {
    const { container } = render(
      <StatisticsWidget
        statsTextClass="mdi text-success"
        bgclassName="dark"
        icon="mdi mdi-cash-multiple"
        description="Description"
        title="Title"
        stats="1234$"
        trend={{
          textClass: 'text-success',
          value: '1234.56',
          time: 'trend',
        }}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('renders as expected without optional params', () => {
    const { container } = render(
      <StatisticsWidget
        description="Description"
        title="Title"
        stats="1234$"
        trend={{
          textClass: 'text-success',
          value: '1234.56',
          time: 'trend',
        }}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
