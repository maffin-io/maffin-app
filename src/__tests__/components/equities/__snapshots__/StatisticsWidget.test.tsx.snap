// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatisticsWidget renders as expected 1`] = `
<div>
  <div
    class="card  rounded-sm p-6"
  >
    <p>
      Title
    </p>
    <p
      class="mdi text-success text-2xl font-semibold my-3"
    >
      1234$
    </p>
    <span
      class="text-sm"
    >
      Description
    </span>
  </div>
</div>
`;

exports[`StatisticsWidget renders as expected without optional params 1`] = `
<div>
  <div
    class="card  rounded-sm p-6"
  >
    <p>
      Title
    </p>
    <p
      class=" text-2xl font-semibold my-3"
    >
      1234$
    </p>
    <span
      class="text-sm"
    >
      Description
    </span>
  </div>
</div>
`;
